src/string_id.h

```diff
@@ -127,6 +127,15 @@ class string_id
          * Returns the actual object this id refers to. May show a debug message if the id is invalid.
          */
         const T &obj() const;
+
+        const T &operator*() const {
+            return obj();
+
+        }
+        const T *operator->() const {
+            return &obj();
+        }
```

Comment:
> providing C++ operators as opposed to our own custom functions improves readability

The problem is, it doesn't. It obfuscates the type system underneath, making ids look like pointers.

> It's a lot less ugly than foo.obj().quux.obj()

Expressions like that SHOULD look ugly, because they are ugly already. Making them use less symbols will not make them less ugly, just make them pretend to be less ugly, hiding the problem.

> I don't want to rewrite the PR and the dependent branches I have for a subjective argument about code beauty when we already have extensive usage of the same idiom (eg. item_location)

I can merge it in the current state if you're willing to take all the blame resulting from confusion regarding the obfuscation here.